test/modellib/ModelLib.hpp

```diff
@@ -97,3 +99,67 @@ bool genLSTMModelAndCompile(
     llvm::SmallVector<int64_t, 3> &hShape,
     llvm::SmallVector<int64_t, 3> &cShape, OMTensor *&wOmt, OMTensor *&rOmt,
     OMTensor *&bOmt, OMTensor *&pOmt);
+
+class ModelLibBuilder {
+public:
+  // Define the model. Subclass should add to the builder all of the specific
+  // parameters that uniquely define the model.
+  ModelLibBuilder(const std::string &sharedLibBaseName);
+  // Destructor needed to free the inputs/outputs data structures.
+  ~ModelLibBuilder();
+  // Build, subclass should generate a graph. If constant nodes are needed by
+  // the model, they should be created here and saved in the subclass, as these
+  // values will be needed to verify the accuracy of the model. The model is
+  // saved in the model and ctx variable.
+  bool build() { llvm_unreachable("subclass must implement build."); }
+  // Compile model from the model and ctx variables. The output is an executable
+  // dynamic library.
+  bool compileAndLoad();
+  // Prepare inputs for running model.
+  bool prepareInputs() { llvm_unreachable("subclass must implement prepare."); }
+  // Run model using prepared inputs, resulting in outputs.
+  bool run();
+  // Verify outputs with reference data.
+  bool verifyOutputs() { llvm_unreachable("subclass must implement verify."); }
```

Comment:
The member functions that should be implemented by derived classes can be declared as pure virtual functions as:
```
virtual bool verifyOutputs() = 0;
```

That way the compiler will know that derived classes must override the pure virtual function at compile time, that is it will not let you instantiate a derived class object unless all the member functions that are pure virtual are defined in the derived class.

Reply:
The disadvantage of virtual member functions is the runtime cost for virtual dispatch. In this case I think that is not important given that this class is for internal testing of models, not something that will penalize the runtime of the `onnx-mlir` compiler.

Reply:
I retooled the constructors to make it easier to understand that it is not to be used in a "virtual" way

Reply:
ok, fully addressed by adding virtual